src/util/ParseableDuration.h

```diff
@@ -29,6 +43,18 @@ class ParseableDuration {
 
     is >> arg;
 
+    try {
+      result = ParseableDuration<DurationType>::fromString(arg);
+    } catch (const std::runtime_error&) {
+      // >> operator does not throw, it sets the error state
+      is.setstate(is.rdstate() | std::ios_base::failbit);
+    }
+
+    return is;
+  }
+
+  static ParseableDuration<DurationType> fromString(std::string_view arg) {
```

Comment:
Add the `// ______` lines or short docstring about all functions that are not super trivial etc. for readability.